src/controllers/preflight.js

```diff
@@ -36,33 +36,71 @@ function PreflightController(ctx, log, env) {
       throw new Error('Invalid request: missing application/json data');
     }
 
-    if (!hasText(data.pageUrl)) {
+    if (!hasText(data.pageUrl?.trim())) {
       throw new Error('Invalid request: missing pageUrl in request data');
     }
+
+    if (!isValidUrl(data.pageUrl)) {
+      throw new Error('Invalid request: invalid pageUrl in request data');
+    }
   }
 
   const createPreflightJob = async (context) => {
-    const { data } = context;
+    const { data, sqs } = context;
 
     try {
       validateRequestData(data);
+    } catch (error) {
+      log.error(`Invalid request data: ${error.message}`);
+      return badRequest(error.message);
+    }
 
+    try {
       const funcVersion = context.func?.version;
       const isDev = /^ci\d*$/i.test(funcVersion);
```

Comment:
this will sadly not work anymore since we migrated to different subscriptions for each env. I added a new secret in AWS that we can use for this:
```suggestion
      const isDev = env.AWS_ENV === 'dev';
```

Reply:
@dzehnder , oh - ok - need to make sure to update any other places that might depend on this logic